src/state/image-dataset/json-dataset/index.ts

```diff
@@ -92,31 +89,111 @@ class JsonRequest implements ImageDataset {
 
         // for now this particular file must be kept up to date.
         // dev-aics-dtp-001/cfedata is the designated repository of internal cfe data sets
+        // this datasets.json contains a list of the names of subdirectories (strings),
+        // each of which is expected to contain a dataset.json file
         this.listOfDatasetsDoc =
-            "http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/datasets.json";
+            "http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/cell-feature-data/datasets.json";
     }
 
     public getAvailableDatasets = () => {
-        return axios
-            .get(`${this.listOfDatasetsDoc}`)
-            .then((metadata: AxiosResponse) => metadata.data);
+        //const megasets: Megaset[] = [];
+        return axios.get(`${this.listOfDatasetsDoc}`).then((metadata: AxiosResponse) => {
+            const datadir = this.listOfDatasetsDoc.substring(
+                0,
+                this.listOfDatasetsDoc.lastIndexOf("/")
+            );
+
+            return Promise.all(
+                (metadata.data as string[]).map((datasetdir) => {
+                    const datasetdirpath = `${datadir}/${datasetdir}`;
+                    const fullurl = `${datasetdirpath}/dataset.json`;
+                    return axios.get(fullurl).then((dataset: AxiosResponse) => {
+                        const megaset = dataset.data as Megaset;
+                        if (megaset.datasets) {
+                            ///////////////////////////
+                            // TODO handle megasets from json flat file data
+                            ///////////////////////////
+                            console.log("has datasets; unexpected; dropping");
+                            return {} as Megaset;
+                        } else {
+                            const topLevelJson = { ...megaset } as unknown as DatasetMetaData;
+                            const megasetInfo = { ...megaset };
+                            megasetInfo.title = topLevelJson.title;
+                            megasetInfo.publications = megaset.publications || [];
+                            megasetInfo.extra = megaset.extra || "";
+                            const id = `${topLevelJson.name}_v${topLevelJson.version}`;
+                            megasetInfo.name = id;
+                            // for single dataset sets we want to store the document with the whole id, to avoid
+                            // grouping versions of the same name together as if they're megasets
+                            megasetInfo.datasets = {
+                                [id]: {
+                                    ...topLevelJson,
+                                    image: `${datasetdirpath}/${topLevelJson.image}`,
```

Comment:
built-in assumption that the image file is local to the current dataset.json file